File: Network/Client/networking_client_test.cc

#include "gtest/gtest.h"

#include "networking_client.h"

class NetworkingClientTest : public ::testing::Test {
public:
   static void cleanupWolfssl(const common::network::IP& ip) {
    NetworkingClient::cleanupWolfssl(ip);
    NetworkingClient::cleanupWolfssl();
  }

};

TEST_F(NetworkingClientTest, SendAsServer) {
  common::network::IP ip;
  ip.set_block("127.0.0.1");
  ip.set_port(8080);

  ASSERT_TRUE(NetworkingClient::setupAsServer("Network/Client/ca.pem", "Network/Client/test-server.pem",
                                  "Network/Client/test.key.pem"));

  common::reactor::Event event;
  event.set_stored_event("some_event");

  // We expect the send to fail because we have no pending client connections.
  ASSERT_FALSE(NetworkingClient::send(ip, event));

  cleanupWolfssl(ip);
}

TEST_F(NetworkingClientTest, SendAsClient) {
  common::network::IP ip;
  ip.set_block("127.0.0.1");
  ip.set_port(8080);

  ASSERT_TRUE(NetworkingClient::setupAsClient("Network/Client/ca.pem"));

  common::reactor::Event event;
  event.set_stored_event("some_event");

  // We expect the send to fail because we have nothing to connect to.
  ASSERT_FALSE(NetworkingClient::send(ip, event));

  cleanupWolfssl(ip);
}

int main(int argc, char **argv) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}

File: javatests/com/keiros/client/network/NetworkingClientTest.java
package com.keiros.client.network;

import static org.junit.Assert.*;

import org.junit.*;

import com.google.protobuf.ByteString;

import com.keiros.common.network.IPClass.IP;
import com.keiros.common.reactor.EventClass.Event;
import com.keiros.client.network.NetworkingClient;
import com.keiros.reactor.base.EventAndMessage;

public class NetworkingClientTest {

    public void before(IP ip) throws Exception {
        NetworkingClient.cleanupWolfssl(ip);
        NetworkingClient.cleanupWolfsslMethod();
    }

    @Test
    public void sendAsServer() throws Exception {
        IP ip = IP.newBuilder().setBlock("127.0.0.1").setPort(8080).build();

        assertTrue(NetworkingClient.setupAsServerMethod("Network/Client/ca.pem", "Network/Client/test-server.pem", "Network/Client/test.key,pem"));

        EventAndMessage eventAndMessage = new EventAndMessage();
        eventAndMessage.message = Event.newBuilder().setStoredEvent(ByteString.copyFromUtf8("something"));

        // We expect this to fail because we have no pending client connections.
        assertFalse(NetworkingClient.send(ip, eventAndMessage));

        before(ip);
    }

    @Test
    public void sendAsClient() throws Exception {
        IP ip = IP.newBuilder().setBlock("127.0.0.1").setPort(8080).build();

        assertTrue(NetworkingClient.setupAsClientMethod("Network/Client/ca.pem"));

        EventAndMessage eventAndMessage = new EventAndMessage();
        eventAndMessage.message = Event.newBuilder().setStoredEvent(ByteString.copyFromUtf8("something"));

        // We expect this to fail because we have no pending client connections.
        assertFalse(NetworkingClient.send(ip, eventAndMessage));

        before(ip);
    }

}
